examples/seismic/tti/wavesolver.py

```diff
@@ -84,7 +78,7 @@ def forward(self, src=None, rec=None, u=None, v=None, m=None,
         # Create the forward wavefield if not provided
         if v is None:
             v = TimeData(name='v', shape=self.model.shape_domain,
-                         save=save, time_dim=self.source.nt,
+                         save=save, time_dim=self.source.nt if save else None,
```

Comment:
same here: should we not compute this information once and for all in the constructor?

Reply:
In this case no, as save is an invocation time parameter, and many tests run through both sides of this conditional with the same `solver` object.